react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ value: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    console.log('State:', this.state);
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <label>
+          Email:
+          <input type="email" value={this.state.value} onChange={this.handleChange} name="email"></input>
```

Comment:
Input tags are self closing, so instead of having a closing `input` element you would do:

```JSX
<input type="email" value={this.state.value} onChange={this.handleChange} name="email" />
```